src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/CosmosFhirDataStore.cs

```diff
@@ -231,9 +238,12 @@ public async Task HardDeleteAsync(ResourceKey key, CancellationToken cancellatio
         {
             EnsureArg.IsNotNull(sqlQuerySpec, nameof(sqlQuerySpec));
 
-            CosmosQueryContext context = new CosmosQueryContext(CollectionUri, sqlQuerySpec, feedOptions);
+            var context = new CosmosQueryContext(CollectionUri, sqlQuerySpec, feedOptions);
 
-            return _cosmosDocumentQueryFactory.Create<T>(DocumentClient, context);
+            using (IScoped<IDocumentClient> documentClient = _documentClientFactory())
```

Comment:
I think `documentClient` should be passed into this method. The parent method should own this scope

Reply:
You are right. I should have caught this. Fixed.